api/datastore/sql/migrations/12_fix_annotations_migrations_route.go

```diff
@@ -0,0 +1,27 @@
+package migrations
+
+import (
+	"context"
+
+	"github.com/fnproject/fn/api/datastore/sql/migratex"
+	"github.com/jmoiron/sqlx"
+)
+
+func up11(ctx context.Context, tx *sqlx.Tx) error {
+	_, err := tx.ExecContext(ctx, "ALTER TABLE routes MODIFY annotations TEXT NULLABLE;")
```

Comment:
pg gives fantastically bad error messages around this, but i believe this (and the same for app) are causing CI to fail with this:

```
    	sql_test.go:94: with migrations now!
    	sql_test.go:97: database is locked
	test.go:29: 
		time="2018-04-20T19:22:19Z" level=info msg="datastore dialed" datastore=postgres max_idle_connections=256
		time="2018-04-20T19:22:19Z" level=info msg="datastore dialed" datastore=postgres max_idle_connections=256
		time="2018-04-20T19:22:19Z" level=error msg="error running migrations" error="error running migration. version: 11 direction: up err: pq: syntax error at or near \"MODIFY\"\npq: current transaction is aborted, commands ignored until end of transaction block"
		time="2018-04-20T19:22:19Z" level=info msg="datastore dialed" datastore=postgres max_idle_connections=256
		time="2018-04-20T19:22:19Z" level=error msg="error running migrations" error="database is locked"
		
	sql_test.go:94: with migrations now!
	sql_test.go:97: database is locked
```

it seems like we need to do, per https://www.postgresql.org/docs/current/static/sql-altertable.html :

```
ALTER TABLE routes ALTER COLUMN annotations DROP NOT NULL;
```

Reply:
so, of course, that doesn't work with mysql... looking into that

Reply:
and sqlite doesn't support this at all, oh boy. at least, these are all getting tossed soon enough. we could just skip it for sqlite for now i guess and leave this NOT NULL, even though it's the one most people use and most of them don't use annotations either. not sure what's best to do exactly, gonna infuse some beans in hot water and think about it